### PR TITLE
Submodules

### PR DESCRIPTION
**Pull request by djaglowski**
## Description of Changes
- Separate executable code and distributable code into modules. 
- Separates most builtin operators into individual modules. This allows each to be imported (or not), which provides more granular control over dependencies that are pulled into another project.

## **Please check that the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] Add a changelog entry (for non-trivial bug fixes / features)
- [ ] CI passes

### CODE REVIEW

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.4483464|-0.4655106|31.519262|-3.556303|
|1|5000|8.310522|-1.1034803|44.836613|-0.19409943|
|1|10000|13.293327|-2.6897383|57.51805|+1.5061989|
|1|50000|61.019127|-11.070168|193.80765|-23.491653|
|1|100000|119.32755|-24.22895|298.73154|-7.6010437|
|10|100|5.3966904|-0.18968916|32.13012|-1.6295776|
|10|500|10.258854|-1.6551533|45.103043|+2.5148163|
|10|1000|17.534882|-1.5689678|58.192215|+2.8138466|
|10|5000|70.25911|-6.614601|170.84576|-7.1726837|
|10|10000|131.02538|-25.087555|300.16177|-32.067078|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.500067|-0.41379|34.389412|-0.6861534|
|1|5000|8.224313|-1.1896896|43.10143|-1.9292831|
|1|10000|13.603823|-2.379243|58.62931|+2.6174583|
|1|50000|63.203632|-8.885662|225.69827|+8.398972|
|1|100000|117.71777|-25.83873|293.60696|-12.725616|
|10|100|5.5346055|-0.051774025|31.43063|-2.3290672|
|10|500|10.103977|-1.81003|41.915947|-0.67227936|
|10|1000|17.845346|-1.258503|59.24677|+3.8684006|
|10|5000|69.44958|-7.4241333|174.53395|-3.484497|
|10|10000|127.83721|-28.275719|345.86758|+13.638733|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.3966382|-0.5172188|33.040276|-2.0352898|
|1|5000|8.379422|-1.0345802|44.27923|-0.7514839|
|1|10000|14.379598|-1.603468|58.427803|+2.4159508|
|1|50000|62.81833|-9.270966|208.8137|-8.485596|
|1|100000|121.74664|-21.80986|294.61234|-11.720245|
|10|100|5.6207705|+0.034390926|33.746094|-0.01360321|
|10|500|11.19005|-0.72395706|46.492863|+3.9046364|
|10|1000|17.086601|-2.0172482|58.032326|+2.6539574|
|10|5000|68.053375|-8.820335|199.91568|+21.897232|
|10|10000|131.92068|-24.192245|321.23816|-10.990692|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.586278|-0.32757902|35.223194|+0.14762878|
|1|5000|8.655376|-0.758626|44.672413|-0.35829926|
|1|10000|14.155467|-1.8275986|57.359646|+1.3477936|
|1|50000|62.484417|-9.604877|190.85129|-26.448013|
|1|100000|118.73801|-24.818497|285.81088|-20.521698|
|10|100|4.896652|-0.6897273|32.81708|-0.9426155|
|10|500|10.516558|-1.3974495|46.041084|+3.452858|
|10|1000|17.293505|-1.8103447|58.018856|+2.6404877|
|10|5000|71.739296|-5.1344147|194.76118|+16.742737|
|10|10000|127.73335|-28.379578|317.7275|-14.501343|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.4139094|-0.49994755|32.07799|-2.9975739|
|1|5000|8.741506|-0.6724968|44.381195|-0.64951706|
|1|10000|13.000135|-2.9829302|56.980873|+0.96902084|
|1|50000|60.315144|-11.774151|207.3327|-9.9665985|
|1|100000|117.73521|-25.821297|293.52734|-12.805237|
|10|100|5.8277636|+0.24138403|33.210262|-0.54943466|
|10|500|11.190045|-0.72396183|46.760777|+4.17255|
|10|1000|18.241589|-0.8622608|58.626617|+3.248249|
|10|5000|70.28071|-6.5930023|199.75606|+21.73761|
|10|10000|128.36255|-27.750381|310.52237|-21.706482|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.3448842|-0.5689728|31.128098|-3.9474678|
|1|5000|8.345244|-1.068758|44.39305|-0.637661|
|1|10000|14.345054|-1.6380119|58.001347|+1.9894943|
|1|50000|61.76367|-10.325623|192.76105|-24.538254|
|1|100000|120.81185|-22.744652|273.8979|-32.434692|
|10|100|5.741564|+0.15518427|29.265356|-4.494341|
|10|500|10.120807|-1.7932005|44.260777|+1.6725502|
|10|1000|17.121326|-1.982523|56.960804|+1.5824356|
|10|5000|68.846504|-8.027206|184.88416|+6.8657074|
|10|10000|126.74345|-29.369484|301.17377|-31.055084|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.2759128|-0.6379442|33.860317|-1.2152481|
|1|5000|8.500121|-0.9138813|41.04701|-3.9837036|
|1|10000|13.448597|-2.5344687|59.543373|+3.5315208|
|1|50000|62.10466|-9.984634|186.51051|-30.788788|
|1|100000|117.22908|-26.327423|292.98532|-13.3472595|
|10|100|5.586354|-2.5749207e-05|30.880926|-2.8787708|
|10|500|9.845055|-2.0689526|42.7465|+0.15827179|
|10|1000|18.569752|-0.5340977|56.97899|+1.6006203|
|10|5000|70.06941|-6.8042984|200.20717|+22.18872|
|10|10000|128.34909|-27.76384|299.95554|-32.273315|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.3276632|-0.5861938|35.19585|+0.120285034|
|1|5000|8.827875|-0.5861273|45.043507|+0.012794495|
|1|10000|13.482998|-2.5000677|61.648975|+5.637123|
|1|50000|61.193005|-10.89629|175.25633|-42.04297|
|1|100000|120.33423|-23.222275|292.43198|-13.900604|
|10|100|5.551884|-0.034495354|31.244747|-2.5149498|
|10|500|9.982941|-1.9310665|43.445312|+0.8570862|
|10|1000|17.483135|-1.6207142|57.030174|+1.6518059|
|10|5000|69.48402|-7.389694|202.22603|+24.20758|
|10|10000|129.1104|-27.002533|297.32098|-34.907867|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.4138808|-0.49997616|32.934|-2.1415672|
|1|5000|8.965715|-0.448287|44.631195|-0.39951706|
|1|10000|14.189954|-1.7931118|58.12783|+2.1159782|
|1|50000|65.81355|-6.2757416|194.45218|-22.847122|
|1|100000|117.70118|-25.855324|289.43628|-16.896301|
|10|100|5.2760987|-0.3102808|32.489494|-1.2702026|
|10|500|10.551985|-1.3620224|44.92147|+2.3332443|
|10|1000|18.638315|-0.4655342|57.41029|+2.0319214|
|10|5000|70.19137|-6.6823425|206.73949|+28.721039|
|10|10000|129.8272|-26.285736|309.73788|-22.490967|

File: .circleci/config.yml [hunk 1]
```diff
@@ -138,7 +138,31 @@ jobs:
       - run:
           name: Unit Tests
           shell: powershell.exe
-          command: go test -race -coverprofile coverage.txt -coverpkg ./... ./...
+          command: >
```
Comment (djaglowski) [inline .circleci/config.yml:141]: I'm certain there is a better way to do this that doesn't involve listing each module manually, but I figure we can handle that later.

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.2759805|-0.6378765|33.557518|-1.5180473|
|1|5000|7.982949|-1.4310536|44.958244|-0.072467804|
|1|10000|14.241764|-1.7413015|58.530174|+2.518322|
|1|50000|60.021988|-12.0673065|190.5505|-26.748795|
|1|100000|116.98474|-26.571762|308.55524|+2.2226562|
|10|100|5.707124|+0.120744705|33.058727|-0.7009697|
|10|500|10.776069|-1.1379385|45.208378|+2.6201515|
|10|1000|17.586666|-1.5171833|56.490707|+1.112339|
|10|5000|69.29502|-7.5786896|179.28946|+1.2710114|
|10|10000|131.76749|-24.345444|319.43253|-12.796326|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|3.2241964|-0.68966055|31.371632|-3.7039337|
|1|5000|8.7932625|-0.62073994|42.780846|-2.2498665|
|1|10000|15.344825|-0.6382408|56.582706|+0.5708542|
|1|50000|60.880787|-11.208508|198.71376|-18.58554|
|1|100000|123.72594|-19.830566|304.00095|-2.3316345|
|10|100|5.0345864|-0.5517931|33.181034|-0.5786629|
|10|500|10.0865135|-1.8274937|42.424706|-0.16352081|
|10|1000|17.086401|-2.0174484|58.53125|+3.1528816|
|10|5000|70.41566|-6.4580536|169.75781|-8.260635|
|10|10000|130.40154|-25.711395|300.7003|-31.528564|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.9827871|+0.051719546|124.77465|-2.4401932|
|1|5000|6.22425|-1.172461|137.79715|+1.208252|
|1|10000|13.379588|-3.034505|143.349|-3.7417908|
|1|50000|67.49333|-11.45668|232.14561|-14.072861|
|1|100000|122.53782|-26.35453|394.29984|-22.594025|
|10|100|2.2241645|-0.5862305|131.00876|+0.2172699|
|10|500|7.465629|-1.2243352|138.40019|+1.3246307|
|10|1000|14.500357|-2.223774|143.73962|-3.0561676|
|10|5000|69.19475|-12.844475|215.08473|-11.018173|
|10|10000|131.48596|-25.274658|363.78204|-13.806732|

**Comment by djaglowski** [comment]
| Log Files | Logs / Second | CPU Avg (%) | CPU Avg Δ (%) | Memory Avg (MB) | Memory Avg Δ (MB) |
| --- | --- | --- | --- | --- | --- |
|1|1000|1.6379627|-0.2931049|126.67174|-0.5431061|
|1|1000|1.8448793|-0.08618832|126.63941|-0.5754318|
|1|5000|5.8449183|-1.5517926|132.46417|-4.1247253|
|1|5000|5.96576|-1.4309506|132.64938|-3.9395142|
|1|10000|12.465878|-3.9482155|143.41446|-3.6763306|
|1|10000|12.258793|-4.1553|141.70352|-5.387268|
|1|50000|57.742924|-21.207088|176.65706|-69.56142|
|1|50000|58.364563|-20.58545|173.75957|-72.45891|
|1|100000|110.69334|-38.199013|223.38133|-193.51253|
|1|100000|113.48622|-35.406128|228.90234|-187.99152|
|10|100|2.362168|-0.44822693|132.0582|+1.2667084|
|10|100|2.3795123|-0.4308827|129.75067|-1.0408173|
|10|500|6.741489|-1.9484754|136.51144|-0.56411743|
|10|500|6.879599|-1.8103652|139.06317|+1.9876099|
|10|1000|13.535056|-3.1890745|144.40544|-2.3903503|
|10|1000|14.345145|-2.3789854|144.04135|-2.7544403|
|10|5000|63.311626|-18.727596|184.14252|-41.96039|
|10|5000|63.48426|-18.554962|178.34537|-47.757538|
|10|10000|121.566216|-35.194405|222.94113|-154.64764|
|10|10000|121.978935|-34.781685|225.87553|-151.71324|

**Review by camdencheek — APPROVED** [review]

**Pull request merged**